src/verifiers/documentStatus/didSigned/didSignedDocumentStatus.ts

```diff
@@ -54,8 +60,23 @@ const verifyV2 = async (document: SignedWrappedDocument<v2.OpenAttestationDocume
       "MISSING_REVOCATION"
     );
 
-  // Support for the NONE method only
-  const revokedOnAny = !revocation.every((r) => r?.type === "NONE");
+  const revocationStatuses: RevocationStatus[] = await Promise.all(
+    revocation.map((r) => {
+      return r?.type === v2.RevocationType.RevocationStore
+        ? isRevokedOnDocumentStore({
+            documentStore: r?.address,
+            merkleRoot,
+            provider: options.provider,
+            targetHash,
+          })
+        : Promise.resolve({
+            revoked: false,
+          });
```

Comment:
This will result in a way to "unrevoke" my document. If I'm a recipient of document and I found out that the issuer revoked my document, I simply obfuscate my revocation type to skip the revocation check. 

Should check that its NONE, else throw. 

Might want to have a separate function to do this on top or something.

Reply:
but wont the previous error (lines 57-61) cover this vulnerability?
the code from 64-74 will only run if r.type is well defined.

separate function as in extract it out as a callback back function to make it neater?

Reply:
got it after discussion with @yehjxraymond